src/classes/CON_Email.cls

```diff
@@ -43,103 +43,184 @@ public class CON_Email {
     */
 	public static void processPreferredEmail(Contact contact, List<Contact> oldlist) {
 
+        Boolean validatePreferred = !UTIL_CustomSettingsFacade.getSettings().Disable_Preferred_Email_Enforcement__c;
+
         // No need to run if we are in a delete context
         if(contact == null) {
             return;
         }
 
 		Boolean isUpdateOrDelete = (oldlist != null) ? true : false;
 
-	    // Store default email field mappings to preferred email pick list.
-	    Map<String,String> preferredEmailMappings = new Map<String,String>{   
-	        'University' => 'UniversityEmail__c',
-	        'Work' => 'WorkEmail__c',
-	        'Alternate' => 'AlternateEmail__c'
-	    };
+        // Build the list of email fields
+        CON_EmailFieldList emailFields = new CON_EmailFieldList(contact);
 
-	    Map<String,String> valuedEmailMap = getEmailFieldsWithValue(contact);
+        if( emailFields.valuedCount() > 0 ) {
 
-        if( valuedEmailMap.size() > 0 ) {
+            // Check the presence of a special cases when running a batch process.
+            // This will prevent some validation errors below to allow batches to complete successfully.
+            if( validatePreferred && System.isBatch() ) {
+                if( !emailFields.valueExists(contact.Email) && String.isBlank(contact.Preferred_Email__c)) {
+                    contact.Preferred_Email__c = 'Email (standard)';
+                }
 
-            // Enforce preferred email field
-            if(String.isBlank(contact.Preferred_Email__c)){
-                contact.addError( Label.PreferredEmailRequiredError );
+                // We are catching a special case here when there are Custom Emails but Email is blank.
+                // If there is only one custom email present, we will set it as preferred. When multiple
+                // custom emails are present the first in the list will be set to preferred.
+                if(String.isBlank(contact.Email) ) {
+                    // Get the first item from the list either one or many
+                    contact.Email = emailFields.valuedFields[0].value;
+                    contact.Preferred_Email__c = emailFields.valuedFields[0].prefLabel;
+                }
             }
 
-            // Get the Api name that matches the value in Preferred Email
-            String preferredApiField = ( preferredEmailMappings.containsKey(contact.Preferred_Email__c) ) ? preferredEmailMappings.get(contact.Preferred_Email__c) : valuedEmailMap.get(contact.Preferred_Email__c);
+            // Enforce preferred email field unless validation is disabled
+            if(validatePreferred && String.isBlank(contact.Preferred_Email__c)){
```

Comment:
This will run for both Batch and non-Batch context. That's the intention, right?